pkg/controllers/provisioning/scheduling/modelinference.go

```
@@ -0,0 +1,181 @@
+		}
+		log.FromContext(ctx).
+			WithValues("Pod", klog.KObj(pod)).
+			V(1).Info(fmt.Sprintf("adding requirements derived from pod's inference flavor %q, %s", flavor.Name, matchExpressions))
+		pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms = append(pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution.NodeSelectorTerms, nodeSelectorTermsCopy...)
+	})
+
+	log.FromContext(ctx).
+		WithValues("Pod", klog.KObj(pod)).
+		V(1).Info(fmt.Sprintf("adding requirements derived from pod's inference flavors, %s", pretty.Concise(pod.Spec.Affinity.NodeAffinity.RequiredDuringSchedulingIgnoredDuringExecution)))
+
+	return nil
+}
+
+func (m *ModelInference) getInferenceFlavors(ctx context.Context, pod *v1.Pod) ([]llmazcoreapi.Flavor, error) {
+	modelName, ok := pod.Labels[llmazcoreapi.ModelNameLabelKey]
+	if !ok {
+		// Ignore the pod that is not created via llmaz's inference service.
+		return nil, nil
+	}
+
+	model := &llmazcoreapi.OpenModel{}
+	if err := m.kubeClient.Get(ctx, types.NamespacedName{Name: modelName}, model); err != nil {
+		return nil, fmt.Errorf("getting open model %q, %w", modelName, err)
+	}
+	modelFlavors := lo.FromPtrOr(model.Spec.InferenceConfig, llmazcoreapi.InferenceConfig{}).Flavors
+
+	serviceFlavorRawStr, ok := pod.Annotations[llmazinferenceapi.InferenceServiceFlavorsAnnoKey]
+	if !ok {
+		// Not all inference pods specify the inference service flavors.
+		return modelFlavors, nil
+	}
+
+	modelFlavorMap := lo.SliceToMap(modelFlavors, func(flavor llmazcoreapi.Flavor) (llmazcoreapi.FlavorName, llmazcoreapi.Flavor) {
+		return flavor.Name, flavor
+	})
+
+	var result []llmazcoreapi.Flavor
+	for _, flavorNameVal := range strings.Split(serviceFlavorRawStr, ",") {
+		flavor, ok := modelFlavorMap[llmazcoreapi.FlavorName(flavorNameVal)]
+		if !ok {
+			return nil, fmt.Errorf("unknown service inference flavor %q", flavorNameVal)
```

Let's just ignore the error here, it won't be destructive to our system.

---

It aligns with https://github.com/InftyAI/scheduler-plugins/blob/main/pkg/plugins/resource_fungibility/resource_fungibility.go#L168. Do you want to change it as well?

---

What would happen if all flavors of the pod were removed from the target model but the model has other flavors? The pod fails to be scheduled or the pod is scheduled to unexpected flavor?

---

Ok, let's be more strict here, we can refactor this part later if needed.